openslides_backend/models/check_models.py

```diff
@@ -0,0 +1,686 @@
+    if value is None:
+        return True
+    if not isinstance(value, list):
+        return False
+    return all([fn(sv) for sv in value])
+
+
+def check_decimal(value: Any) -> bool:
+    if value is None:
+        return True
+    if isinstance(value, str):
+        pattern = r"^-?(\d|[1-9]\d+)\.\d{6}$"
+        if re.match(pattern, value):
+            return True
+    return False
+
+
+def check_json(value: Any, root: bool = True) -> bool:
+    if value is None:
+        return True
+    if not root and (isinstance(value, int) or isinstance(value, str)):
+        return True
+    if isinstance(value, list):
+        return all(check_json(x, root=False) for x in value)
+    if isinstance(value, dict):
+        return all(check_json(x, root=False) for x in value.values())
+    return False
+
+
+class Checker:
+    def __init__(self, data: Dict[str, List[Any]], is_import: bool = False) -> None:
+        self.data = data
+
+        self.models: Dict[str, Type["Model"]] = {
+            collection.collection: model_registry[collection]
+            for collection in model_registry
+        }
+
+        if is_import:
+            self.modify_models_for_import()
+
+        self.errors: List[str] = []
+
+        self.template_prefixes: Dict[
+            str, Dict[str, Tuple[str, int, int]]
+        ] = defaultdict(dict)
+        self.generate_template_prefixes()
+
+    def get_fields(self, collection: str) -> Iterable[Field]:
+        return self.models[collection]().get_fields()
+
+    def modify_models_for_import(self) -> None:
+        collection_allowlist = (
+            "user",
+            "meeting",
+            "group",
+            "personal_note",
+            "tag",
+            "agenda_item",
+            "list_of_speakers",
+            "speaker",
+            "topic",
+            "motion",
+            "motion_submitter",
+            "motion_comment",
+            "motion_comment_section",
+            "motion_category",
+            "motion_block",
+            "motion_change_recommendation",
+            "motion_state",
+            "motion_workflow",
+            "motion_statute_paragraph",
+            "poll",
+            "option",
+            "vote",
+            "assignment",
+            "assignment_candidate",
+            "mediafile",
+            "projector",
+            "projection",
+            "projector_message",
+            "projector_countdown",
+            "chat_group",
+        )
+        for collection in list(self.models.keys()):
+            if collection not in collection_allowlist:
+                del self.models[collection]
+        # TODO: mediafile blob handling.
+
+    def generate_template_prefixes(self) -> None:
+        for collection in self.models.keys():
+            for field in self.get_fields(collection):
+                if not isinstance(field, BaseTemplateField):
+                    continue
+                field_name = field.get_structured_field_name("")
```

Comment:
You should be able to use `BaseTemplateField.index` instead of splitting the structured field by yourself.

Reply:
I use the Field function here, it should be okay.

Reply:
```suggestion
                field_name = field.get_template_field_name()
```

Reply:
Replaced.